### PR TITLE
Improved SToRM32 support

### PR DESCRIPTION
**Pull request by pistolero**
Added reading of current angles and sending them back in MOUNT_STATUS msg.

### CODE REVIEW

**Comment by rmackay9** [comment]
Well, that's interesting.  I guess this actually replaces the MAVLink interface previously used and replaces it with the storm serial interface.  That's not really the direction we were moving in but your change does have some nice improvements in it.
I'll have a chat with some other devs on the dev call tomorrow to see what we want to do about this.

**Comment by pistolero** [comment]
@rmackay9 I found no way to read angles back via SToRM32 mavlink interface, and have decided to change code mimic the way how Alexmos gimbals are handled.

File: libraries/AP_Mount/AP_Mount_SToRM32.cpp [hunk 1]
```diff
@@ -3,24 +3,29 @@
...
+    _reply_counter(0),
+    _reply_type(ReplyType_UNKNOWN)
 {}
 
 // init - performs any required initialisation for this instance
 void AP_Mount_SToRM32::init(const AP_SerialManager& serial_manager)
 {
     // get_mavlink_channel for MAVLink2
-    if (serial_manager.get_mavlink_channel(AP_SerialManager::SerialProtocol_MAVLink, 1, _chan)) {
+    if ((_port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_SToRM32, 0))) {
```
Comment (magicrub) [inline libraries/AP_Mount/AP_Mount_SToRM32.cpp]: Please don't have assignments within an if statement

File: libraries/AP_Mount/AP_Mount_SToRM32.cpp [hunk 2]
```diff
@@ -82,8 +89,24 @@ void AP_Mount_SToRM32::update()
     }
 
     // resend target angles at least once per second
-    if (resend_now || ((hal.scheduler->millis() - _last_send) > AP_MOUNT_STORM32_RESEND_MS)) {
-        send_do_mount_control(ToDeg(_angle_ef_target_rad.y), ToDeg(_angle_ef_target_rad.x), ToDeg(_angle_ef_target_rad.z), MAV_MOUNT_MODE_MAVLINK_TARGETING);
+    resend_now = resend_now || ((hal.scheduler->millis() - _last_send) > AP_MOUNT_STORM32_RESEND_MS);
```
Comment (magicrub) [inline libraries/AP_Mount/AP_Mount_SToRM32.cpp]: Another assignment within an if

**Comment by rmackay9** [comment]
@pistolero,
We had a chat on the dev call including Luis and decided we will enable both the existing MAVLink and your new serial protocol driver.  This gives us two solutions and we can see which one is more popular (probably your serial one).
I've modified your patches a bit so that the two drivers can co-exist.
https://github.com/rmackay9/rmackay9-ardupilot/tree/pistolero-storm1

One problem I've found is that if I try to control the gimbal with my transmitters ch6 tuning knob it seems to act more like a rate controller rather than an angle controller.  I'm wondering if maybe we are not passing the desired angles correctly to the storm gimbal using this new protocol?

**Comment by pistolero** [comment]
@rmackay9 that's strange it worked like angle controller in my tests. Will double check.

**Comment by rmackay9** [comment]
@pistolero,
Ok, I think I figured out the issue, the angle requests were being multiplied by 100 unnecessarily.  When I remove this x100 it all works fine.  I wonder if it depends upon the version of the storm gimbal.  I'm using version 067e.  It looks like there are new vesions (0.70 is the latest) so maybe I should switch to use that?

**Comment by pistolero** [comment]
@rmackay9 that's weird. It was not working for me without that multiplication. I'm not sure which version I had. 

I don't have gimbal at hands right now, so it would be nice if you could upgrade to latest firmware and test it again.

**Comment by rmackay9** [comment]
@pistolero, ok I've just retested with SToRM32 ver 0.7 and it seems happier with the fixed angles.

It's going through the automated software testing now but once that's done I'll push this into master.

**Comment by pistolero** [comment]
@rmackay9 ok, good to know. Maybe we need to add a note somewhere that it is adviced to use latest firmware (0.7+) with that code.

**Comment by rmackay9** [comment]
merged, thanks!

**Pull request closed without merging**